protocol/messenger_sync_contact_request_decision_test.go

```diff
@@ -0,0 +1,115 @@
+package protocol
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/requests"
+	"github.com/status-im/status-go/protocol/tt"
+)
+
+type MessengerSyncContactRequestDecisionSuite struct {
+	MessengerBaseTestSuite
+	m2 *Messenger
+}
+
+func TestMessengerSyncContactRequestDecision(t *testing.T) {
+	suite.Run(t, new(MessengerSyncContactRequestDecisionSuite))
+}
+
+func (s *MessengerSyncContactRequestDecisionSuite) SetupTest() {
+	s.MessengerBaseTestSuite.SetupTest()
+
+	m2, err := newMessengerWithKey(s.shh, s.privateKey, s.logger, nil)
+	s.Require().NoError(err)
+	s.m2 = m2
+
+	PairDevices(&s.Suite, m2, s.m)
+	PairDevices(&s.Suite, s.m, m2)
+}
+
+func (s *MessengerSyncContactRequestDecisionSuite) TearDownTest() {
+	s.Require().NoError(s.m2.Shutdown())
+	s.MessengerBaseTestSuite.TearDownTest()
+}
+
+func (s *MessengerSyncContactRequestDecisionSuite) createUserB() *Messenger {
+	key, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+	userB, err := newMessengerWithKey(s.shh, key, s.logger, nil)
+	s.Require().NoError(err)
+	return userB
+}
+
+func (s *MessengerSyncContactRequestDecisionSuite) TestSyncAcceptContactRequest() {
+	userB := s.createUserB()
+	defer func() {
+		s.Require().NoError(userB.Shutdown())
+	}()
+
+	// send contact request to m/m2, m and m2 are paired
+	request := &requests.AddContact{ID: common.PubkeyToHex(&s.m2.identity.PublicKey)}
+	_, err := userB.AddContact(context.Background(), request)
+	s.Require().NoError(err)
+
+	// check m and m2 received contact request
+	var contactRequestMessageID types.HexBytes
+	_, err = WaitOnMessengerResponse(s.m, func(r *MessengerResponse) bool {
+		if len(r.ActivityCenterNotifications()) > 0 {
+			for _, n := range r.ActivityCenterNotifications() {
+				if n.Type == ActivityCenterNotificationTypeContactRequest {
+					contactRequestMessageID = n.ID
+					return true
+				}
+			}
+		}
+		return false
```

Comment:
I'd appreciate inverting the condition for better readability 
```suggestion
		if len(r.ActivityCenterNotifications()) == 0 {
		    return false
		}
	    for _, n := range r.ActivityCenterNotifications() {
			if n.Type == ActivityCenterNotificationTypeContactRequest {
			    contactRequestMessageID = n.ID
				return true
			}
		}
		return false
```

Reply:
Oh, it's even better actually.
The condition is not needed at all, the loop will simply make 0 iterations.